src/EditorFeatures/Core/Implementation/Classification/CopyPasteAndPrintingClassificationBufferTaggerProvider.Tagger.cs

```diff
@@ -104,10 +104,14 @@ private TagSpanIntervalTree<IClassificationTag> CachedTags
 
             private void OnEventSourceChanged(object sender, TaggerEventArgs _)
             {
-                _owner._notificationService.RegisterNotification(
-                    OnEventSourceChanged_OnForeground,
-                    _owner._asyncListener.BeginAsyncOperation("SemanticClassificationBufferTaggerProvider"),
-                    _cancellationTokenSource.Token);
+                _owner.ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    using var _ = _owner._asyncListener.BeginAsyncOperation("SemanticClassificationBufferTaggerProvider");
+
+                    await Task.Delay(15, _cancellationTokenSource.Token).ConfigureAwait(false);
+                    await _owner.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_cancellationTokenSource.Token);
```

Comment:
teh 15 ms was hard coded into the notification service.  not sure the best way to translate that over here.  i"ve just put the 15 in at all the callsites.  we could also introduce an extension fort his pattern.  i.e. `ThreadingContext.SwitchToMainThreadAfterDelay(15, ...)`

Reply:
The delay was used to allow for operation batching, but we don't do any batching now. I think it can be removed.